src/plugins/ui_stateful/clipboard.ts

```diff
@@ -184,6 +187,16 @@ export class ClipboardPlugin extends UIPlugin {
           this.status = "invisible";
         }
         break;
+      case "KEEP_PAINT_FORMAT": {
+        this.keepPaintingFormat = true;
+        break;
+      }
+      case "CANCEL_PAINT_FORMAT": {
+        this._isPaintingFormat = false;
```

Comment:
I'd add a condition in the allowDispatch to check if we are currently in a paint format. Not a real problem in practice, but it doesn't hurt.